src/modules/storage/BlockStorage.ts

```diff
@@ -0,0 +1,128 @@
+
+/**
+ * The class that insert and read the block into the database.
+ */
+class BlockStorage
+{
+    /**
+     *  The instance of sqlite3
+     */
+    private db: sqlite.Database;
+
+    /**
+     * Constructor
+     * @param filename Valid values are filenames,
+     * ":memory:" for an anonymous in-memory database and
+     * an empty string for an anonymous disk-based database
+     * @param callback : If provided, this function will be called when
+     * the database was opened successfully or when an error occurred.
+     * The first argument is an error object.
+     */
+    constructor (filename : string, callback : any = null)
+    {
+        this.db = new sqlite.Database(filename,
+            sqlite.OPEN_CREATE | sqlite.OPEN_READWRITE |
+            sqlite.OPEN_SHAREDCACHE);
+        this.db.configure("busyTimeout", 1000);
+        this.createTable(callback);
+    }
+
+    /**
+     * Creates a table. table name is `block`.
+     * @param callback If provided, this function will be called when
+     * the database was finished successfully or when an error occurred.
+     * The first argument is an error object.
+     */
+    public createTable (callback : any = null)
+    {
+        var sql =
+        `CREATE TABLE IF NOT EXISTS block
+        (
+            height INTEGER PRIMARY KEY,
+            prev_block TEXT,
+            merkle_root TEXT,
+            signature TEXT
+        )`;
+        this.db.run(sql, (err : any) =>
+        {
+            if (callback != null)
+                callback(err);
+        });
+    }
+
+    /**
+     * Puts a block to database
+     * @param data a block data
+     * @param callback If provided, this function will be called when
+     * the database was finished successfully or when an error occurred.
+     * The first argument is an error object.
+     */
+    public put (data : any, callback : any = null)
+    {
+        var sql =
+        `INSERT INTO block
+            (height, prev_block, merkle_root, signature)
+        VALUES
+            (?, ?, ?, ?)
+        ON CONFLICT(height) DO UPDATE SET
+            prev_block = prev_block,
+            merkle_root = merkle_root,
+            signature = signature`;
+        this.db.run(sql,
+            [
+                data.header.height.value,
+                data.header.prev_block,
+                data.header.merkle_root,
+                data.header.signature
+            ], (err : any) =>
+        {
+            if (callback != null)
+                callback(err);
+        });
+    }
+
+    /**
+     * Gets a block data
+     * @param height the height of the block to get
+     * @param callback If provided, this function will be called when
+     * the database was finished successfully or when an error occurred.
+     * The first argument is an error object.
+     * The second argument is result set.
+     */
+    public get (height : number, callback : any = null)
+    {
+        this.db.all("SELECT * FROM block WHERE height = ?",
```

Comment:
We'd better define the column and bring it in.
Not `SELECT *`
Unnecessary information may be added when returning to API after inquiry.

Reply:
It would be better if we expand it in the future.

Reply:
Good point!

Reply:
I changed it.